src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -1415,6 +2045,54 @@ export default class PlotsTab extends React.Component<IPlotsTabProps,{}> {
         }
     }
 
+    @computed get horzLogScaleFunction():IAxisLogScaleParams|undefined {
+        return this.makeAxisLogScaleFunction(this.horzSelection);
+    }
+
+    @computed get vertLogScaleFunction():IAxisLogScaleParams|undefined {
+        return this.makeAxisLogScaleFunction(this.vertSelection);
+    }
+
+    private makeAxisLogScaleFunction(axisSelection:AxisMenuSelection):IAxisLogScaleParams|undefined {
+
+        const MIN_LOG_ARGUMENT = 0.01;
+
+        if (!axisSelection.logScale) {
+            return undefined;
+        }
+
+        let label;          // suffix that will appear in the axis label
+        let fLogScale;     // function for (log-)transforming a value
+        let fInvLogScale;  // function for back-transforming a value transformed with fLogScale
+
+        // log-transformation parameters for non-treatment reponse 
+        // profile data Note: log2-transformation is used by default
+        if(axisSelection.dataType !== AlterationTypeConstants.TREATMENT_RESPONSE) {
+            
+            label = "log2";
+            fLogScale = (x:number, offset:number) => Math.log2(Math.max(x, MIN_LOG_ARGUMENT));
+            fInvLogScale = (x:number) => Math.pow(2, x);
+            
+        // log-transformation parameters for treatment reponse
```

Comment:
I think it's clearer to put these comments inside the branches, this one in particular I find really confusing

Reply:
Comments moved. Do you also object to the content of the comment in the sense that it is unclear?

Reply:
No, I think the content of the comment is good. Thanks!